programs/clearing_house/src/math/oracle.rs

```diff
@@ -46,14 +76,88 @@ pub fn get_oracle_status<'a>(
         oracle_mark_spread_pct,
         is_valid: oracle_is_valid,
         mark_too_divergent: is_oracle_mark_too_divergent,
+        oracle_validity,
     })
 }
 
+pub fn oracle_validity(
+    last_oracle_twap: i128,
+    oracle_price_data: &OraclePriceData,
+    valid_oracle_guard_rails: &ValidityGuardRails,
+) -> ClearingHouseResult<OracleValidity> {
+    let OraclePriceData {
+        price: oracle_price,
+        confidence: oracle_conf,
+        delay: oracle_delay,
+        has_sufficient_number_of_data_points,
+        ..
+    } = *oracle_price_data;
+
+    let is_oracle_price_nonpositive = oracle_price <= 0;
+    if is_oracle_price_nonpositive {
+        msg!("Invalid Oracle: Non-positive (oracle_price <=0)");
+    }
+
+    let is_oracle_price_too_volatile = ((oracle_price
```

Comment:
can probs do oracle_price.max(last_oracle_twap) / oracle_price.min(last_oracle_twap) > ratio and save the || i think?